xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/displayers/BaseDisplayer.vue

```diff
@@ -27,63 +27,97 @@
+  >
+    <template #trigger>
+
+      <div
+        :class="{ view: isView, edit: !isView }"
+        ref="displayerRoot"
+        @click="showPopover = !showPopover"
+        @dblclick="setEdit"
+        @keypress.self.enter="setEdit"
+      >
         <!--
-          Default Editor widget
-          Normally this should rarely be used, as the default displayer
-          should provide a default editor if no displayer is specified
-          However, this is useful if a custom displayer only implement
-          its Viewer widget, as a default Editor widget would still be provided
+          The base displayer contains three slots: `viewer`, `editor`, and `loading`.
+          It displays `viewer` or `loading` according to its current state: `this.isView` when `this.isLoading` is
+          false, and `loading` otherwise.
         -->
-        <input
-          class="default-input"
-          type="text"
-          size="1"
-          v-autofocus
-          v-model="baseValue"
-        />
-      </slot>
-    </div>
 
-    <slot name="loading" v-if="isLoading">
-      <XWikiLoader></XWikiLoader>
-    </slot>
+        <!-- The slot containing the displayer Viewer widget -->
+        <div tabindex="0" v-if="isView && !isLoading">
+          <slot name="viewer">
+            <!--
+              Default Viewer widget
+              Normally this should rarely be used, as the default displayer
+              should provide a default viewer if no displayer is specified
+              However, this is useful if a custom displayer only implement
+              its Editor widget, as a default Viewer widget would still be provided
+            -->
+            <span>{{ value }}</span>
+          </slot>
+        </div>
+
+        <!-- The slot containing the displayer Editor widget -->
+        <div @keypress.enter="applyEdit"
+             @keydown.esc="cancelEdit"
+             v-if="!isView && !isLoading"
+             tabindex="0"
+             ref="editBlock"
+        >
+          <slot name="editor">
+            <!--
+              Default Editor widget
+              Normally this should rarely be used, as the default displayer
+              should provide a default editor if no displayer is specified
+              However, this is useful if a custom displayer only implement
+              its Viewer widget, as a default Editor widget would still be provided
+            -->
+            <input
+              class="default-input"
+              type="text"
+              size="1"
+              v-autofocus
+              v-model="baseValue"
+            />
+          </slot>
+        </div>
 
-  </div>
+        <slot name="loading" v-if="isLoading">
+          <XWikiLoader></XWikiLoader>
+        </slot>
+      </div>
+    </template>
+
+    <!-- Popover content -->
+    <template>
+      <div class="displayer-action-list">
+        <ActionEdit
+          :displayer="{
+            isEditable,
+            setEdit: () => void setEdit()
```

Comment:
Out of curiosity, why is it needed to prefix the method call with `void`?

Reply:
I might have forgotten to remove it after some code change. Basically void returns `undefined` whatever comes next, it can be useful when we don't want a function return value to be returned by the caller function.
But here `setEdit` already returns `undefined` so this is not needed.

Reply:
We need to agree on code style with @mflorea but I find it clearer to write.
```javascript
setEdit: () => { setEdit(); }
```